src/main/scala/com/amazon/deequ/metrics/Metric.scala

```diff
@@ -37,13 +39,24 @@ trait Metric[T] {
   def flatten(): Seq[DoubleMetric]
 }
 
+/**
+ * Full-column metrics store the entire column of row-level pass/fail results
+ */
+trait FullColumn {
+  val fullColumn: Option[Column] = None
+
+  def sum(colA: Option[Column], colB: Option[Column]): Option[Column] =
+    if (colA.equals(colB)) colA else None
```

Comment:
The implementation of this function does not seem intuitive. Can we add comments what the function is doing?